newIDE/app/src/UI/CompactSemiControlledNumberField/index.js

```diff
@@ -48,6 +48,8 @@ type Props = {|
   useLeftIconAsNumberControl?: boolean,
   renderEndAdornmentOnHover?: (className: string) => React.Node,
   onClickEndAdornment?: () => void,
+  formatValueOnFocus?: string => string,
+  formatValueOnBlur?: string => string,
```

Comment:
added those 2 new props that allow formatting a value when the input is not focused.
Here this is helpful for adding the % and removing it.
I suppose this will be helpful in the future too.

Reply:
This could may be renamed to `formatDisplayedValue...` as it's literally just a display thing

Reply:
Yeah, "formatDisplayedValue" sounds cleaner!